internal/services/services.go

```diff
@@ -46,6 +69,73 @@ func (dm *defaultMetricsService) AllCounters(f func(string, int64)) {
 	dm.storage.AllCounters(f)
 }
 
+func (dm *defaultMetricsService) Save(m *model.Metrics) (*model.Metrics, error) {
+	switch metricType := m.MType; metricType {
+	case "counter":
+		if strings.TrimSpace(m.ID) == "" {
+			return nil, ErrIDRequired
+		}
+		if m.Delta == nil {
+			return nil, ErrDeltaRequired
+		}
+		newValue := dm.saveCounter(m.ID, *m.Delta)
+		retVal := &model.Metrics{}
+		retVal.ID = m.ID
+		retVal.MType = m.MType
+		retVal.Delta = &newValue
+		return retVal, nil
+	case "gauge":
+		if strings.TrimSpace(m.ID) == "" {
+			return nil, ErrIDRequired
+		}
+		if m.Value == nil {
+			return nil, ErrValueRequired
+		}
+		retVal := &model.Metrics{}
+		dm.SaveGauge(m.ID, *m.Value)
+		newValue := *m.Value
+		retVal.ID = m.ID
+		retVal.MType = m.MType
+		retVal.Value = &newValue
+		return retVal, nil
+	default:
+		return nil, ErrInvalidMType
+	}
+}
+
+func (dm *defaultMetricsService) Value(m *model.Metrics) (*model.Metrics, error) {
+	result := &model.Metrics{}
+	result.ID = m.ID
+	result.MType = m.MType
+	switch m.MType {
+	case "counter":
+		if delta, ok := dm.GetCounter(m.ID); ok {
+			result.Delta = &delta
+			return result, nil
+		} else {
+			return nil, ErrIDNotFound
+		}
```

Comment:
аналогично в других местах
```suggestion
		return nil, ErrIDNotFound
		
```